cmds/clients/contestcli/transport/http/http.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package http
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+
+	"github.com/facebookincubator/contest/pkg/api"
+	"github.com/facebookincubator/contest/pkg/types"
+	"github.com/facebookincubator/contest/plugins/listeners/httplistener"
+
+	flag "github.com/spf13/pflag"
+	"github.com/insomniacslk/xjson"
+)
+
+var (
+	flagAddr = flag.StringP("addr", "a", "http://localhost:8080", "ConTest server [scheme://]host:port[/basepath] to connect to")
+)
+
+// HttpPartiallyDecodedResponse is a httplistener.HTTPAPIResponse, but with the Data not fully decoded yet
+type HttpPartiallyDecodedResponse struct {
+	ServerID string
+	Type  string
+	Data  json.RawMessage
+	Error *xjson.Error
+}
+
+// Http communicates with ConTest Server via http(s)/json transport
+// Http implements the Transport interface
+type Http struct {
+}
+
+func (h *Http) Version(requestor string) (*api.VersionResponse, error) {
+	resp, err := h.request(requestor, "version", url.Values{})
+	if err != nil  {
+		return nil, err
```

Comment:
seems like we could have some easy coverage wins here by adding a bunch of invalid argument cases to tests

Reply:
I am slightly surprised by this warning, isn't this entire file untouched?

Reply:
maybe. anyway, feel free to ignore this for now.